server/app/services/ti/TrustedIntermediaryService.java

```diff
@@ -251,49 +255,103 @@ public Form<EditTiClientInfoForm> updateClientInfo(
+      SearchParameters searchParameters,
+      ImmutableList<AccountModel> allAccounts,
+      List<SearchParameters.ParamTypes> missingParams) {
+    Optional<LocalDate> maybeDOB =
+        validateAndConvertSearchParamDOB(searchParameters, missingParams);
     return allAccounts.stream()
         .filter(
             account ->
                 ((account.newestApplicant().get().getApplicantData().getDateOfBirth().isPresent()
-                        && searchParameters.dateQuery().isPresent()
-                        && !searchParameters.dateQuery().get().isEmpty()
+                        && maybeDOB.isPresent()
                         && account
                             .newestApplicant()
                             .get()
                             .getApplicantData()
                             .getDateOfBirth()
                             .get()
-                            .equals(
-                                dateConverter.parseIso8601DateToLocalDate(
-                                    searchParameters.dateQuery().get())))
-                    || (searchParameters.nameQuery().isPresent()
-                        && !searchParameters.nameQuery().get().isEmpty()
+                            .equals(maybeDOB.get()))
+                    || (!missingParams.contains(SearchParameters.ParamTypes.NAME)
                         && account
                             .getApplicantName()
                             .toLowerCase(Locale.ROOT)
                             .contains(
                                 searchParameters.nameQuery().get().toLowerCase(Locale.ROOT)))))
         .collect(ImmutableList.toImmutableList());
   }
+
+  private static List<SearchParameters.ParamTypes> findMissingSearchParams(
+      SearchParameters searchParameters) {
+    List<SearchParameters.ParamTypes> missing = new ArrayList<>();
+
+    /* Filtering to transform empty strings into empty optionals.
+    Empty parameters should return all accounts. */
+    Optional<String> filteredNameQuery =
+        searchParameters.nameQuery().filter(s -> !s.isEmpty() && !s.isBlank());
+    Optional<String> filteredDayQuery =
+        searchParameters.dayQuery().filter(s -> !s.isEmpty() && !s.isBlank());
+    Optional<String> filteredMonthQuery =
+        searchParameters.monthQuery().filter(s -> !s.isEmpty() && !s.isBlank());
+    Optional<String> filteredYearQuery =
+        searchParameters.yearQuery().filter(s -> !s.isEmpty() && !s.isBlank());
+
+    if (filteredNameQuery.isEmpty()) {
+      missing.add(SearchParameters.ParamTypes.NAME);
+    }
+    if (filteredDayQuery.isEmpty()) {
+      missing.add(SearchParameters.ParamTypes.DAY);
+    }
+    if (filteredMonthQuery.isEmpty()) {
+      missing.add(SearchParameters.ParamTypes.MONTH);
+    }
+    if (filteredYearQuery.isEmpty()) {
+      missing.add(SearchParameters.ParamTypes.YEAR);
+    }
+
+    return missing;
+  }
+
+  public static boolean validateSearch(SearchParameters searchParameters) {
```

Comment:
Please revert static. Also any reason why is it public?

Reply:
This is `public static` so that it can be called as a utility function from the `TrustedIntermediaryDashboardView.java`.